source/ncengine/graphics2/diligent/resource/PostProcessColorSinkBufferResource.cpp

```diff
@@ -31,14 +32,26 @@ auto PostProcessColorSinkBufferResource::Add(Diligent::IRenderDevice& device,
         return addedIndices;
     }
 
-    if (numColorRenderTargets + m_colorRenderTargets.size() > m_maxTextures)
+    if (numSamples > 1)
```

Comment:
This function does this check a few times and basically has duplicated code each time, just targeting different vectors. Maybe something like this with a function/lamba to create it based on sample count would cut things down a bit:
```cpp
struct RTStuf
{
    std::vector<Diligent::RefCntAutoPtr<Diligent::ITexture>& x;
    std::vector<Diligent::IDeviceObject*>& y;
    std::vector<Diligent::IDeviceObject*>& z;
};
```